File: 2025_3/pr_12924.js
function solution(n) {
    let ans=0
    let total=0
    let left = 0
    for(let i=1; i<=n; i++) {
        total += i
        
        while (total > n){
            total -= left
            left +=1
        }
        
        if (total==n) {
            ans += 1
        }
        
    }
    return ans
}

File: 2025_3/pr_12924.py
def solution(n):
    ans = 0
    #연속된 자연수들로 표현
    for i in range(1,n+1):
        total = 0
        for num in range(i, n+1):
            total += num
            if total == n:
                ans += 1
                break
            elif total > n:
                break
    return ans
        
